src/components/Dropdown/Dropdown.tsx

```diff
@@ -103,7 +118,11 @@ export function Dropdown(props: PropsWithChildren<DropdownProps>): JSX.Element {
     if (e.key === 'ArrowDown') {
       updateFocusedItem(focusedIndex + 1);
     } else if (e.key === 'ArrowUp') {
-      updateFocusedItem(focusedIndex - 1);
+      if (alwaysSelectOption && value !== focusedValue) {
+        updateFocusedItem(focusedIndex - 2);
+      } else {
+        updateFocusedItem(focusedIndex - 1);
+      }
     } else if (e.key === 'Tab' && !e.shiftKey) {
```

Comment:
if a user typed something and tab out, this wouldn't be selecting anything at the moment. We may need to update this part as well to select the first option or something

Reply:
best to ask products thought on this, perhaps it would just be clearing the input text and not select anything

Reply:
chatted with Cindy, wouldn't select anything. Ack